### PR TITLE
Remove duplicated Transfer-Encoding header

### PR DESCRIPTION
**Pull request by Shiranuit**
## What does this PR do ?

- Fix an incorrect behaviour where the Transfer-Encoding header was duplicated because we were adding it ourselves and so was doing uWebsocket, this wasn't noticed because we are using NGINX which was sanitizing the HTTP responses, hiding the issue from us.
- Add `Jest` to create functional tests without cucumber
- Add functional tests that verifies that the headers are sent properly

### CODE REVIEW

**Comment by kuzzle** [comment]
## Executed Checks

### :heavy_check_mark: Branch Merge Check



### :heavy_check_mark: Changelog Tag Check



### :heavy_check_mark: PR Body Check


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 59027e3f0281ad558222d1966c1176875f34d0d4
</p>

File: lib/core/network/protocols/httpwsProtocol.js
Comment (alexandrebouthinon) [inline lib/core/network/protocols/httpwsProtocol.js:790]: Don't you want to keep the Content-Length header management?

Comment (Shiranuit) [reply lib/core/network/protocols/httpwsProtocol.js:790]: No it's also set by uWebsocket when writing with `tryEnd`

Comment (alexandrebouthinon) [reply lib/core/network/protocols/httpwsProtocol.js:790]: Ok didn't know about it 🙂

**Review by alexandrebouthinon — APPROVED** [review]

File: package.json
Comment (alexandrebouthinon) [inline package.json:17]:
```suggestion
    "test:jest": "npm run jest test-functional",
```

File: package.json
Comment (alexandrebouthinon) [inline package.json:16]: You should take a look to the jest.config to only test `function-tests` folder for now

File: package.json
Comment (alexandrebouthinon) [inline package.json:23]:
```suggestion
    "test:functional": "npm run test:functional:http && npm run test:functional:websocket && npm run test:jest",
```

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

File: package.json
Comment (alexandrebouthinon) [reply package.json:16]: For now all the unit tests are launched with this command

Comment (Shiranuit) [reply package.json:16]: The command `jest:test` only launches tests within the `test-functional` folder
The command `jest` is here so you can launch test individualy using `npm run jest <file>` if you only want to execute a specific test

File: test-functional/network/http/http-stream.test.ts
Comment (Aschen) [inline test-functional/network/http/http-stream.test.ts:1]: On other project, we name the jest directory `tests`, I think we should stick to this

Comment (Shiranuit) [reply test-functional/network/http/http-stream.test.ts:1]: I know but we have a folder named `test` so naming another folder `tests` only for test done with Jest doesn't seems right

**Review by alexandrebouthinon — APPROVED** [review]

**Comment by Shiranuit** [comment]
Pull request blocked because of an error within the Javascript SDK, waiting for it to be merged
https://github.com/kuzzleio/sdk-javascript/pull/720

File: test-functional/network/http/http-stream.test.ts
Comment (Aschen) [reply test-functional/network/http/http-stream.test.ts:1]: Maybe we can name it `jest` ?

**Review by rolljee — APPROVED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Pull request merged**
